nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryDatabaseTable.java

```diff
@@ -433,7 +434,7 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
         } finally {
             session.commit();
             try {
-                // Update the state
+                // Check if we should, and then update the state
```

Comment:
Looks like maybe too much was removed? The comment is still changed but the AtomicBoolean is gone, also I couldn't see a reference to where rollbackStateChanges() is called.

Reply:
Fixed.

Reply:
Odd. Will fix.

Reply:
OK, try it now.